src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,35 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 public class ReadWriteList<E> {
     private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final Lock readLock = lock.readLock();
+    private final Lock writeLock = lock.writeLock();
 
     public void add(E element) {
-        // write your code here
+        try {
+            writeLock.lock();
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        try {
+            readLock.lock();
+            return list.get(index);
+        } finally {
+            readLock.unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
```

Comment:
should this be locked as well?

Reply:
After thinking about it, I think that a lock is needed here. Thank you for your comments